### PR TITLE
Updated Pytorch, Tensorflow and GCC module pages

### PR DESCRIPTION
**Pull request by lkr-ind**
This PR addresses two concerns

1. For users who want to have both tenorflow and pytorch in the same environment, I have added details on how to do that. This is because if we try to install one after another, it leads to errors. 

2. I have also added details about GCCcore module and what issues it causes if try to use it.

### CODE REVIEW

File: docs/hpc/applications/guides/pytorch.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 ## Conda
 Whether using  CPU's or GPU's, you should install PyTorch into a clean Anaconda environment. You will need to first setup up your own [Anaconda environment](./conda.md).
 
-Please take care when installing PyTorch with other packages. It isn't uncommon for PyTorch to conflict with other packages so we generally recommend keeping your PyTorch environment to a minimum.
+Please take care when installing PyTorch with other packages. It isn't uncommon for PyTorch to conflict with other packages so we generally recommend keeping your PyTorch environment to a minimum. Also, for users who are interested to have both the Pytorch and TensorFlow installed in the same environment, please replace the last line in the code block  with the one provided in the TensorFlow and Pytorch section.
```
Comment (davidhemp) [inline docs/hpc/applications/guides/pytorch.md:8]:
```suggestion
Please take care when installing PyTorch with other packages. It isn't uncommon for PyTorch to conflict with other packages so we generally recommend keeping your PyTorch environment to a minimum. Also, for users who are interested in having both Pytorch and TensorFlow installed in the same environment, please replace the last line in the code block with the one provided in the TensorFlow and Pytorch section.
```

File: docs/hpc/applications/guides/pytorch.md [hunk 2]
```diff
@@ -17,10 +17,22 @@ mkdir -p $CONDA_PREFIX/etc/conda/activate.d
...
+
+Please make sure that you have run all the commands (except the last one as mentioned in the comments) from the above section before running the following command.
+
+```bash
+python3 -m pip install torch torchvision torchaudio tensorflow==2.12.*
+```
+
+## Check Pytorch Installation
+
+You should then be able to set this work in a job, for example
```
Comment (davidhemp) [inline docs/hpc/applications/guides/pytorch.md:35]:
```suggestion
You should then be able to test this works in a job, for example:
```

**Review by davidhemp — COMMENTED** [review]
Just a few typos (I think one was originally mine).

**Comment by lkr-ind** [comment]
@davidhemp  Thanks for your review and suggestions. I have accepted your suggestions and committed them to the code. I hope this can now be merged.

**Pull request merged**